primitives/runtime/src/offchain/storage_lock.rs

```diff
@@ -337,6 +355,17 @@ impl<'a, 'b, L: Lockable> StorageLockGuard<'a, 'b, L> {
 	pub fn forget(mut self) {
 		let _ = self.lock.take();
 	}
+
+	/// Extend the lock by guard if the lock is existed.
+	///
+	/// Can be used to extend deadline when doing iterations
+	/// that it's hard to be estimated consumed time. After
+	/// a successful step, extend the lock if the lock is still active.
+	pub fn extend_lock(&mut self) {
+		if let Some(ref mut lock) = self.lock {
```

Comment:
I think this should return information whether extension was successful, so one can react on that.

Reply:
Agree with this one. Currently you might be calling `extend_lock` in a loop, thinking that it's still acquired, while in reality it is just going to do nothing.
What's even worse you can have such sequence:
1. Lock is acquired by Worker 1
2. Lock reaches deadline, but Worker 1 still holds the `Guard`
3. Lock is acquired by Worker 2
4. Lock is extended by Worker 1 (but the owner technically is worker 2)

Of course this was an issue with previous API (that merely holding a `Guard` does not guarantee the lock being still active), but with `extend_lock` pattern I feel like we should give some extra guarantees, cause it suggests that it's okay to hold the lock for a longer periods of time and extend it from time to time (Without it was clear from the design that lock is meant to be held for some critical section that should complete fast).